docs/design/node_agent_config-design.md

```diff
@@ -0,0 +1,91 @@
+# NodeAgentConfig configuration with restic/kopia
+Date: 2023-07-25
+
+## Abstract
+
+New configuration for the file system backup & restore that will be used by the OADP and allow to choose Restic or Kopia uploader type.
+
+## Background
+
+For the file system backup and restore the Velero may use Restic or Kopia as the uploader mechanism. A number of tests were [performed](https://velero.io/docs/main/performance-guidance/) by the Velero community to compare those mechanisms. In many cases Kopia uploader is a much better performing mechanism and as such should be added to the OADP operator.
+
+## Goals
+
+- A new option `nodeagentconfig` to allow configuration of restic or kopia uploader
+- Backwards compatibility with the current OADP configuration schema options, namely `restic`
+- Preparation for the future deprecation of the current `restic` configuration option (from OADP 1.4+)
+- Allow new schema to be used by the datamover node agent
+
+## Non Goals
+- Enablment of datamover node agent
+- Deprecation of the "restic" configuration option
+- E2E tests for the `kopia` or `restic` uplaoder
+
+## High-Level Design
+
+Since new `nodeagentconfig` configuration option is a sibling of the `restic` one, the `restic` will be used as part of the new `nodeagentconfig` with addition of one `UploaderType` option that will be either `kopia` or `restic`.
+
+Without that option a default `restic` will be used. That default option may change in the future.
+
+## Detailed Design
+
+A new structure will be added, that includes the ResticConfig inline and extends this with one new parameter `UploaderType`:
+
+```go
+type NodeAgentConfig struct {
+	// Embedding ResticConfig
+	// +kubebuilder:validation:Optional
+	ResticConfig `json:",inline"`
```

Comment:
This have to be ResticConfig, because we are using fields from the `ResticConfig struct` inline (same as copy-paste of those fields in that particular line). I just want those to not duplicate same code in 2 places.

Reply:
@mpryc @shubham-pampattiwar As an alternative, we could  create a new `NodeAgentFields` struct that is identical to current `ResticConfig` (i.e. everything but the uploader-type) and this can be inlined by both `ResticConfig` and `NodeAgentConfig`.

We wouldn't want to call it `UploaderConfig` in any case because:
1) It doesn't include the `uploaderType` field.
2) We need this for datamover too, not just uploader/fsBackup, since to enable datamover you must include a `NodeAgentConfig`.

Reply:
`NodeAgentConfig` makes more sense.

Reply:
I am closing this comment as it's the same as other thread. Currently we will have `NodeAgentCommonFields`, which is exactly the same structure as `ResticConfig` that will be included inline by `NodeAgentConfig` and `ResticConfig`. From `ResticConfig` we will remove all fields.